### PR TITLE
Add support to add/modify challenges

### PR DESCRIPTION
**Pull request by flacial**
Related to #2081 

# Problem

In order to make the challenges tab/page, the `AdminLessonInputs` component has to support adding and updating challenges.

# Solution

Adds support for adding and updating challenges by conditionally setting the hooks and values based on the flag `isModule`.

This PR doesn't make the `AdminLessonInputs` generic but adds support for challenges atop of modules.

Notes:

- In a future PR, this component will become generic.

### CODE REVIEW

File: components/admin/lessons/AdminLessonInputs/AdminLessonInputs.tsx [hunk 1]
```diff
@@ -18,36 +18,50 @@ import {
 } from '@apollo/client'
 
 type Module = { id: number; name: string; content: string; order: number }
+type Challenge = {
```
Comment (anthonykhoa) [inline components/admin/lessons/AdminLessonInputs/AdminLessonInputs.tsx:21]: There should already be a `Challenge` type set somewhere that you can use

Comment (flacial) [reply components/admin/lessons/AdminLessonInputs/AdminLessonInputs.tsx:21]: I'll look into it.

**Review by SlyBouhafs — APPROVED** [review]

**Comment by flacial** [comment]
This page needs compete refactoring. The code isn't easy to read nor understand.

**Review by songz — APPROVED** [review]

**Comment by flacial** [comment]
Closing this to work on it again.

**Pull request closed without merging**